src/python/pants/jvm/test/junit_test.py

```diff
@@ -606,6 +612,178 @@ class SimpleTest {
+            "SimpleTest.java": dedent(
+                """
+                package org.pantsbuild.example;
+
+                import junit.framework.TestCase;
+                import java.nio.file.Files;
+                import java.nio.file.Path;
+
+                public class SimpleTest extends TestCase {
+                   public void testHello() throws Exception {
+                        assertEquals("lol ducks", Files.readString(Path.of("ducks.txt")));
+                   }
+                }
+                """
+            ),
+            "ducks.txt": "lol ducks",
+        }
+    )
+
+    test_result = run_junit_test(rule_runner, "example-test", "SimpleTest.java")
+
+    assert test_result.exit_code == 0
+    assert re.search(r"Finished:\s+testHello", test_result.stdout) is not None
+    assert re.search(r"1 tests successful", test_result.stdout) is not None
+    assert re.search(r"1 tests found", test_result.stdout) is not None
+
+
+@maybe_skip_jdk_test
+def test_vintage_files_dependencies(rule_runner: RuleRunner) -> None:
+    rule_runner.write_files(
+        {
+            "3rdparty/jvm/default.lock": JUNIT4_RESOLVED_LOCKFILE.serialize(
+                [ArtifactRequirement(coordinate=JUNIT_COORD)]
+            ),
+            "BUILD": dedent(
+                f"""\
+                jvm_artifact(
+                  name = 'junit_junit',
+                  group = '{JUNIT_COORD.group}',
+                  artifact = '{JUNIT_COORD.artifact}',
+                  version = '{JUNIT_COORD.version}',
+                )
+                junit_tests(
+                    name='example-test',
+                    dependencies= [
+                        ':junit_junit',
+                        ':ducks',
+                    ],
+                )
+                files(
+                    name="ducks",
+                    sources="*.txt",
+                )
+                """
+            ),
+            "SimpleTest.java": dedent(
+                """
+                package org.pantsbuild.example;
+
+                import junit.framework.TestCase;
+                import java.nio.file.Files;
+                import java.nio.file.Path;
+
+                public class SimpleTest extends TestCase {
+                   public void testHello() throws Exception {
+                        assertEquals("lol ducks", Files.readString(Path.of("ducks.txt")));
+                   }
+                }
+                """
+            ),
+            "ducks.txt": "lol ducks",
+        }
+    )
+
+    test_result = run_junit_test(rule_runner, "example-test", "SimpleTest.java")
+
+    assert test_result.exit_code == 0
+    assert re.search(r"Finished:\s+testHello", test_result.stdout) is not None
+    assert re.search(r"1 tests successful", test_result.stdout) is not None
+    assert re.search(r"1 tests found", test_result.stdout) is not None
+
+
+@pytest.mark.skip  # TODO(14537) `relocated_files` doesn't presently work, un-skip when fixing that.
```

Comment:
```suggestion
@pytest.mark.skip("TODO(14537) `relocated_files` doesn't presently work, un-skip when fixing that.")
```

Hm, yea: inspecting this, I don't see any reason why it shouldn't already be working. But given the further work that will be happening with codegen (and it being slightly off topic) deferring it makes good sense.